cnxpublishing/views/admin.py

```diff
@@ -522,3 +526,169 @@ def admin_content_status_single_POST(request):
             args['response'] = title + " set to bake!"
 
     return args
+
+
+@view_config(route_name='admin-featured-books', request_method='GET',
+             renderer='templates/featured-books.html',
+             permission='view')
+def admin_featured_books(request):
+    settings = request.registry.settings
+    with db_connect() as db_conn:
+        with db_conn.cursor() as cursor:
+            cursor.execute("""
+                SELECT ident_hash(f.uuid, f.major_version, f.minor_version),
+                (SELECT name
+                from modules as m
+                WHERE f.uuid=m.uuid
+                    and (f.major_version=m.major_version OR
+                         f.major_version is NULL)
+                    and (f.minor_version=m.minor_version OR
+                         f.minor_version is NULL)
+                ORDER by revised DESC LIMIT 1)
+                FROM featured_books as f;""")
+            data = cursor.fetchall()
+            featured_books = []
+            for row in data:
+                title = row[1]
+                if title is None:
```

Comment:
Hmm, good to play it safe, but we have a `NOT NULL` on the `name` column of `modules`, where `title` came from, so unnecessary complexity here.